From d0365e43906bbfa168df5c3ba8756932b81f044b Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Mon, 18 Apr 2022 11:53:18 -0500
Subject: [PATCH 1/1] ARM:dts:aspeed: Modify all device tree to enable MCTP over PCIe

MCTP enabled on all platform dts files.
aspeed-g6.dtsi: MCTP and PCIe configuration added

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts     |  4 ++
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts   |  4 ++
 arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts     |  4 ++
 arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts |  4 ++
 arch/arm/boot/dts/aspeed-g6.dtsi              | 71 +++++++++++++++++++
 .../interrupt-controller/aspeed-scu-irq.h     | 16 +++++
 6 files changed, 103 insertions(+)
 create mode 100644 include/dt-bindings/interrupt-controller/aspeed-scu-irq.h

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index 16a028f63d87..d03fc4c8225f 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -71,6 +71,10 @@
 	};
 };

+&mctp0 {
+	status	= "okay";
+};
+
 &mdio0 {
 	status = "okay";

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index dd2efb7356d5..31272f6f992c 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -70,6 +70,10 @@
 	};
 };

+&mctp0 {
+	status	= "okay";
+};
+
 &mdio0 {
 	status = "okay";

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
index 43a6eca1c844..82d7cbb62037 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
@@ -71,6 +71,10 @@
 	};
 };

+&mctp0 {
+	status	= "okay";
+};
+
 &mdio0 {
 	status = "okay";

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts b/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
index 65adea9cdf0a..5b8c6c8594f4 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
@@ -69,6 +69,10 @@
 	};
 };

+&mctp0 {
+	status	= "okay";
+};
+
 &mdio0 {
 	status = "okay";

diff --git a/arch/arm/boot/dts/aspeed-g6.dtsi b/arch/arm/boot/dts/aspeed-g6.dtsi
index f7f36098a9d1..5924542a5001 100644
--- a/arch/arm/boot/dts/aspeed-g6.dtsi
+++ b/arch/arm/boot/dts/aspeed-g6.dtsi
@@ -3,6 +3,7 @@

 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/interrupt-controller/aspeed-scu-ic.h>
+#include <dt-bindings/interrupt-controller/aspeed-scu-irq.h>
 #include <dt-bindings/clock/ast2600-clock.h>

 / {
@@ -100,6 +101,11 @@
 			    <0x40466000 0x2000>;
 			};

+		ahbc: ahbc@1e600000 {
+			compatible = "aspeed,aspeed-ahbc", "syscon";
+			reg = < 0x1e600000 0x100>;
+		};
+
 		fmc: spi@1e620000 {
 			reg = < 0x1e620000 0xc4
 				0x20000000 0x10000000 >;
@@ -357,6 +363,71 @@
 				quality = <100>;
 			};

+			pcie_ep:pcie_ep@1e6ed000 {
+				compatible = "aspeed,ast2600-pcie-ep", "syscon";
+				reg = <0x1e6ed000 0x100>;
+
+				resets = <&syscon ASPEED_RESET_PCIE_DEV_O>;
+				rc_offset = <0x80>;
+				msi_address = <0x1e770058>;
+
+				status = "disabled";
+			};
+
+			pcie: pcie@1e6ed200 {
+				compatible = "aspeed,ast2600-pcie", "syscon";
+				reg = <0x1e6ed200 0x100>;
+				device_type = "pci";
+				bus-range = <0x00 0xff>;
+
+				#address-cells = <3>;
+				#size-cells = <2>;
+				ranges = <0x01000000 0 0x00010000 0x00010000 0x0 0x10000   /* I/O */
+					  0x02000000 0x0 0x70000000 0x70000000 0 0x0fffffff>; /* memory */
+				#interrupt-cells = <1>;
+
+				interrupt-map-mask = <0 0 0 7>;
+				interrupt-map = <0 0 0 1 &pcie_intc0 0>, /* INT A */
+						<0 0 0 2 &pcie_intc0 1>, /* INT B */
+						<0 0 0 3 &pcie_intc0 2>, /* INT C */
+						<0 0 0 4 &pcie_intc0 3>; /* INT D */
+
+				resets = <&syscon ASPEED_RESET_PCIE_RC_O>;
+				rc_offset = <0xC0>;
+				msi_address = <0x1e77005C>;
+				aspeed,ahbc = <&ahbc>;
+				status = "disabled";
+
+				pcie_intc0: legacy-interrupt-controller {
+					interrupt-controller;
+					#interrupt-cells = <1>;
+					interrupt-parent = <&gic>;
+				};
+			};
+
+			mctp0: mctp@1e6e8000 {
+				compatible = "aspeed,ast2600-mctp";
+				reg = <0x1e6e8000 0x30>;
+				interrupts-extended = <&gic GIC_SPI 26 IRQ_TYPE_LEVEL_HIGH>, <&scu_ic0 ASPEED_AST2600_SCU_IC0_PCIE_PERST_LO_TO_HI>;
+				interrupt-names = "mctp", "pcie";
+				resets = <&syscon ASPEED_RESET_DEV_MCTP>;
+				aspeed,scu = <&syscon>;
+				aspeed,pcieh = <&pcie_ep>;
+				status = "disabled";
+			};
+
+			mctp1: mctp@1e6f9000 {
+				compatible = "aspeed,ast2600-mctp";
+				reg = <0x1e6f9000 0x30>;
+				interrupts-extended = <&scu_ic0 ASPEED_AST2600_SCU_IC0_PCIE_RCRST_LO_TO_HI>;
+				interrupt-names = "pcie";
+				pcie_rc;
+				resets = <&syscon ASPEED_RESET_RC_MCTP>, <&syscon ASPEED_RESET_DEV_MCTP>;
+				aspeed,scu = <&syscon>;
+				aspeed,pcieh = <&pcie>;
+				status = "disabled";
+			};
+
 			xdma: xdma@1e6e7000 {
 				compatible = "aspeed,ast2600-xdma";
 				reg = <0x1e6e7000 0x100>;
diff --git a/include/dt-bindings/interrupt-controller/aspeed-scu-irq.h b/include/dt-bindings/interrupt-controller/aspeed-scu-irq.h
new file mode 100644
index 000000000000..09bb7ebaf85d
--- /dev/null
+++ b/include/dt-bindings/interrupt-controller/aspeed-scu-irq.h
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier: (GPL-2.0+ OR MIT) */
+#ifndef DT_BINDINGS_ASPEED_SCU_IRQ_H
+#define DT_BINDINGS_ASPEED_SCU_IRQ_H
+
+#define ASPEED_SCU_VGA_CURSOR_CHANGE        0
+#define ASPEED_SCU_VGA_SCRATCH_REG_CHANGE   1
+#define ASPEED_SCU_PCIE_REST_LOW_TO_HIGH    2
+#define ASPEED_SCU_PCIE_REST_HIGH_TO_LOW    3
+#define ASPEED_SCU_LPC_REST_LOW_TO_HIGH     4
+#define ASPEED_SCU_LPC_REST_HIGH_TO_LOW     5
+#define ASPEED_SCU_ISSUE_MSI                6
+
+/* ast2600 scu2 */
+#define ASPEED_SCU2_LPC_REST_LOW_TO_HIGH    4
+#define ASPEED_SCU2_LPC_REST_HIGH_TO_LOW    5
+#endif
--
2.17.1

